### ISSUE TITLE
A subquery indicates TranslateApplyRule calculating the domain in a wrong way.

### ISSUE THREAD

**Issue by chenzl25** [risingwavelabs/risingwave#6522]
### Describe the bug

We will miss some row because we calculate the domain in a wrong way.

### To Reproduce

```sql 
create table t1 (a int, b int);
create table t2 (c int, d int);
insert into t1 values (1, 1);
select * from t1 left join t2 on t1.a = t2.c where exists (select t2.d is null);
 a | b | c | d
---+---+---+---
(0 rows)
```

### Expected behavior

```
 a | b | c | d
---+---+---+---
 1 | 1 |   |
(1 row)
```

**Comment by jon-chuang** [comment]
Fixed by: https://github.com/risingwavelabs/risingwave/issues/6499?

**Comment by chenzl25** [comment]
> Fixed by: #6499?

Not really, we have a special way to calculate the domain of the apply previously which only can be used to SPJ form sql, but even in this restricted form, they can not calculate the correct domain. I think we need to use general way to calculate the domain. Directly calculate the domain could lead to cross join, if we don't push any predicates down first. I think we need to find a way deal with them properly without sacrificing performance compared with the current domain calculation method.
